src/neuroconv/tools/nwb_helpers/_models/_base_dataset_models.py

```diff
@@ -0,0 +1,131 @@
+"""Base Pydantic models for DatasetInfo and DatasetConfiguration."""
+from typing import Any, Dict, Tuple, Union
+from abc import abstractmethod, ABC
+
+import h5py
+import numcodecs
+import numpy as np
+from pydantic import BaseModel, Field, root_validator, validator
+
+
+class DatasetInfo(BaseModel):
+    # TODO: When using Pydantic v2, replace with
+    # model_config = ConfigDict(allow_mutation=False)
+    class Config:  # noqa: D106
+        allow_mutation = False
+        arbitrary_types_allowed = True
+
+    object_id: str = Field(description="The UUID of the neurodata object containing the dataset.")
+    location: str = Field(description="The relative location of the this dataset within the in-memory NWBFile.")
+    full_shape: Tuple[int, ...] = Field(description="The maximum shape of the entire dataset.")
+    dtype: np.dtype = Field(  # TODO: When using Pydantic v2, replace np.dtype with InstanceOf[np.dtype]
```

Comment:
This is intentional - this is really a machine generated container intended to be returned by a function, I would never expect or want a user to hand type an instance of it (especially the object ID), or if they did for whatever reason I would avoid duck typing because that is WAY harder to validate in JSON because I'd have to compile a string listing of all allowed inputs to Dtype (if you know an easy way to get that I'm all ears)

Reply:
Got you. 
We don't leave that level of generality so why pay for it? If our use case is solved by only using np.dtype then we should just use that.